model.py

```diff
@@ -585,11 +575,167 @@ class Place(Base):
+        geographic comparisons will fail.
+        """
+        place, is_new = get_one_or_create(
+            _db, Place, type=self.EVERYWHERE,
+            create_method_kwargs=dict(external_id="Everywhere",
+                                      external_name="Everywhere")
+        )
+        return place
+
+    @classmethod
+    def larger_place_types(cls, type):
+        """Return a list of place types known to be bigger than `type`.
+
+        Places don't form a strict heirarchy. In particular, ZIP codes
+        are not 'smaller' than cities. But counties and cities are
+        smaller than states, and states are smaller than nations, so
+        if you're searching inside a state for a place called "Japan",
+        you know that the nation of Japan is not what you're looking
+        for.
+        """
+        larger = [Place.EVERYWHERE]
+        if type not in (Place.NATION, Place.EVERYWHERE):
+            larger.append(Place.NATION)
+        if type in (Place.COUNTY, Place.CITY, Place.POSTAL_CODE):
+            larger.append(Place.STATE)
+        if type == Place.CITY:
+            larger.append(Place.COUNTY)
+        return larger
+    
+    @classmethod
+    def parse_name(cls, place_name):
+        """Try to extract a place type from a name.
+
+        :return: A 2-tuple (place_name, place_type)
+
+        e.g. "Kern County" becomes ("Kern", Place.COUNTY)
+        "Arizona State" becomes ("Arizona", Place.STATE)
+        "Chicago" becaomes ("Chicago", None)
+        """
+        check = place_name.lower()
+        place_type = None
+        if check.endswith(' county'):
+            place_name = place_name[:-7]
+            place_type = Place.COUNTY
+
+        if check.endswith(' state'):
+            place_name = place_name[:-6]
+            place_type = Place.STATE
+        return place_name, place_type
+
+    @classmethod
+    def lookup_by_name(cls, _db, name, place_type=None):
+        """Look up one or more Places by name.
+        """
+        if not place_type:
+            name, place_type = cls.parse_name(name)
+        qu = _db.query(Place).outerjoin(PlaceAlias).filter(
+            or_(Place.external_name==name, Place.abbreviated_name==name,
+                PlaceAlias.name==name)
+        )
+        if place_type:
+            qu = qu.filter(Place.type==place_type)
+        return qu
+
+    @classmethod
+    def name_parts(cls, name):
+        """Split a nested geographic name into parts.
+
+        "Boston, MA" is split into ["MA", "Boston"]
+        "Lake County, Ohio, USA" is split into
+        ["USA", "Ohio", "Lake County"]
+
+        There is no guarantee that these place names correspond to
+        Places in the database.
+
+        :param name: The name to split into parts.
+        :return: A list of place names, with the largest place at the front
+           of the list.
+        """
+        return [x.strip() for x in reversed(name.split(",")) if x.strip()]
+    
+    def strictly_intersects(self, qu):
```

Comment:
I would call this "strictly_inside". "strictly_intersects" sounds like the opposite of this to me.